File: Java/0MVC-Java-Android-Retrofit/src/com/example/mvc_java_android_retrofit/MainActivity.java
package com.example.mvc_java_android_retrofit;

import io.reactivex.Observable;
import android.os.Bundle;
import android.app.Activity;
import android.view.Menu;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.adapter.rxjava2.*;

//Retrofit requires at minimum Java 7 or Android 2.3.

//https://www.vogella.com/tutorials/Retrofit/article.html
//https://blog.matheuscastiglioni.com.br/consumindo-web-service-no-android-com-retrofit/
public class MainActivity extends Activity {

	public ApiInterface retrofitApi;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		
		Retrofit retrofit = new Retrofit.Builder()
	    .baseUrl("http://javarestjson.herokuapp.com/api")
	    .addConverterFactory(GsonConverterFactory.create())
	    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
	    .build();
		
		retrofitApi = retrofit.create(ApiInterface.class);
		
		//https://www.baeldung.com/retrofit-rxjava
		Call<Repo> call = retrofitApi.listRepository(null);
		call.enqueue(new Callback<Repo>() {
			@Override
			public void onFailure(Call<Repo> arg0, Throwable arg1) {
				
			}

			@Override
			public void onResponse(Call<Repo> arg0, Response<Repo> arg1) {
				
			}
		});
		
		
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

}

File: Kotlin/MVPKotlinAndroidWearable/mobile/src/main/java/com/example/mvp_kotlin_android_wearable/MainActivity.kt
package com.example.mvp_kotlin_android_wearable
package com.example.mvp_kotlin_android_wearable.com.example.model

import com.example.mvp_kotlin_android_wearable.com.example.model.UsersList

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.google.android.gms.common.api.Api

import retrofit2.Callback
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    val BASE_URL:String = "http://javarestjson.herokuapp.com/api/produtos"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loadSQL()
    }

    // function to call server and update ui
    fun loadSQL() {
        var retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        var api = retrofit.create(Api::class.java)
        var call = api.users
        call.enqueue(object : Callback&lt; UsersList&gt; {

            override fun onResponse(call: Call&lt;UsersList&gt;?, response: Response&lt;UsersList&gt;?) {
            var ures = response?.body()
            var user = ures?.users
            var length = user!!.size

            for (i in 0 until length) {
                str = str + "\n" + user.get(i).id + " " + user.get(i).login
            }

            //tv_user!!.text = str
        }

            override fun onFailure(call: Call&lt;UsersList&gt;?, t: Throwable?) {
            Log.v("Error", t.toString())
        }
        })
    }
}
